src/QuartoNotebookWorker/src/QuartoNotebookWorker.jl

```diff
@@ -0,0 +1,87 @@
+module QuartoNotebookWorker
+
+walk(x, _, outer) = outer(x)
+walk(x::Expr, inner, outer) = outer(Expr(x.head, map(inner, x.args)...))
+postwalk(f, x) = walk(x, x -> postwalk(f, x), f)
+
+module Packages
+
+import QuartoNotebookWorker: postwalk
+
+import TOML
+
+is_precompiling() = ccall(:jl_generating_output, Cint, ()) == 1
+
+const packages = let
+    project = TOML.parsefile(joinpath(@__DIR__, "..", "Project.toml"))
+    uuid = Base.UUID(project["uuid"])
+    key = "packages"
+    is_precompiling() && Base.record_compiletime_preference(uuid, key)
+    Base.get_preferences(uuid)[key]
+end
+const rewrites = Set(Symbol.(first.(splitext.(basename.(packages)))))
+
+function rewrite_import_or_using(expr::Expr)
+    return postwalk(expr) do ex
+        if Meta.isexpr(ex, :(.))
+            root = get(ex.args, 1, nothing)
+            root in rewrites && prepend!(ex.args, [fullname(@__MODULE__)..., root])
+        end
+        ex
+    end
+end
+
+is_include(ex) =
+    Meta.isexpr(ex, :call) &&
+    length(ex.args) == 2 &&
+    ex.args[1] == :include &&
+    isa(ex.args[2], String)
+
+function rewrite_include(ex::Expr)
+    insert!(ex.args, 2, rewriter)
```

Comment:
what's happening here, are you adding a function to the ast so that the `include` calls change to those versions that map the included ast first? probably better to explain that in a comment, took me a moment to understand